pkg/arvo/neo/cod/std/src/fil/axal.txt

```diff
@@ -0,0 +1,338 @@
+  |=  pax=pith
+  ^+  fat
+  ?~  pax  [~ kid.fat]
+  =/  kid  (~(get by kid.fat) i.pax)
+  ?~  kid  fat
+  fat(kid (~(put by kid.fat) i.pax $(fat u.kid, pax t.pax)))
+```
+
+Returns an axal with the node at path `pax` removed.
+
+## `+dip`
+
+Descend to the axal at this path, [which does what]
+
+```
+++  dip
+  |=  pax=pith
+  ^+  fat
+  ?~  pax  fat
+  =/  kid  (~(get by kid.fat) i.pax)
+  ?~  kid  [~ ~]
+  $(fat u.kid, pax t.pax)
+```
+
+Returns an axal whose uppermost node is the node at the given path `pax`.
+
+## `+fit`
+
+Fetch file at the longest existing prefix of this path.
+
+```
+++  fit
+  |=  pax=pith
+  ^+  [pax fil.fat]
+  ?~  pax  [~ fil.fat]
+  =/  kid  (~(get by kid.fat) i.pax)
+  ?~  kid  [pax fil.fat]
+  =/  low  $(fat u.kid, pax t.pax)
+  ?~  +.low
+    [pax fil.fat]
+  low
+```
+
+Returns a `(pair pith *)` where `*` is the node at the given path `pax`.
+
+## `+gas`
+
+Merge a list of `(pair pith *)` with the axal.
+
+[??? that's what it looks like this should do but i'd expect a `|-` in here]
+
+```
+++  gas
+  |*  lit=(list (pair pith _?>(?=(^ fil.fat) u.fil.fat)))
+  ^+  fat
+  ?~  lit  fat
+  $(fat (put p.i.lit q.i.lit), lit t.lit)
+```
+
+Returns an axal.
+
+## `+get`
+
+Get a node.
+
+```
+++  get
+  |=  pax=pith
+  fil:(dip pax)
+```
+
+Returns a unit of the node.
+
+## `+got`
+
+Get a node, error if the node doesn't exist.
+
+```
+++  got
+  |=  pax=pith
+  ~|  missing-room/pax
+  (need (get pax))
+```
+
+Returns the node or errors.
+
+## `+gut`
+
+Get a node, with a default to fall back on.
+
+```
+++  gut
+  |*  [pax=pith dat=*]
+  =>  .(dat `_?>(?=(^ fil.fat) u.fil.fat)`dat, pax `pith`pax)
+  ^+  dat
+  (fall (get pax) dat)
+```
+
+Returns the node if it exists, return the default noun `dat` if not.
+
+## `+has`
+
+Check if the axal contains a node as this path.
+
+```
+++  has
+  |=  pax=pith
+  !=(~ (get pax))
+```
+
+Returns `%.y` if the node exists, `%.n` if not.
+
+## `+kid`
+
+???
+
+```
+++  kid
+  |=  pax=pith
+  ^-  (map pith _?>(?=(^ fil.fat) u.fil.fat))
+  =.  fat  (dip pax)
+  =.  fat  snip
+  =.  fil.fat  ~
+  tar
+```
+
+Returns a `(map pith *)` where `*` is the type of the node in the sample axal.
+
+## `+kids`
+
+Get the kids of the node at the given path.
+
+```
+++  kids
+  |=  pax=pith
+  ^-  (axil _?>(?=(^ fil.fat) u.fil.fat))
+  :-  (get pax)
+  (kid pax)
+```
+
+Returns an [`axil`](https://docs.urbit.org/language/hoon/reference/arvo#axil).
+
+[why an axil? not sure what the different usage is in this case]
+
+## `+lop`
+
+Recursively remove nodes from the tree. [??? - need to play around with this but i think the idea is that if given path /foo/bar/baz it will remove the nodes at addresses /foo, /bar, and /baz]
+
+```
+++  lop
+  |=  pax=pith
+  ^+  fat
+  ?~  pax  fat
+  |-
+  ?~  t.pax  fat(kid (~(del by kid.fat) i.pax))
+  =/  kid  (~(get by kid.fat) i.pax)
+  ?~  kid  fat
+  fat(kid (~(put by kid.fat) i.pax $(fat u.kid, pax t.pax)))
+```
+
+Returns an axal.
+
+## `+parent`
+
+???
```

Comment:
walks backwards up the axal, starting at a child node specified by `cur` and returns a `(unit pith)` of the closest populated ancestor